src/components/Tabs/tests/Tabs.test.tsx

```diff
@@ -193,4 +206,245 @@ describe('<Tabs />', () => {
+          key: 'ArrowRight',
+        });
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowLeft',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+    });
+
+    describe('ArrowUp', () => {
+      it('shifts focus to the last tab when pressing ArrowUp', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} selected={0} />,
+        );
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowUp',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(2);
+      });
+
+      it('shifts focus to the first tab when pressing ArrowUp on the second tab', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} />,
+        );
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowRight',
+        });
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowRight',
+        });
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowLeft',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+    });
+  });
+
+  describe('getVisibleAndHiddenTabIndices()', () => {
+    it('properly sets getVisibleAndHiddenTabIndices', () => {
+      const mockTabs = [
+        {content: 'Tab 1', id: 'tab-1'},
+        {content: 'Tab 2', id: 'tab-2'},
+        {content: 'Tab 3', id: 'tab-3'},
+        {content: 'Tab 4', id: 'tab-4'},
+        {content: 'Tab 5', id: 'tab-5'},
+        {content: 'Tab 6', id: 'tab-6'},
+      ];
+      const selected = 0;
+      const disclosureWidth = 50;
+      const tabWidths = [82, 160, 150, 100, 80, 120];
+      const containerWidth = 300;
+      const actualIndices = getVisibleAndHiddenTabIndices(
+        mockTabs,
+        selected,
+        disclosureWidth,
+        tabWidths,
+        containerWidth,
+      );
+      const expectedIndices = {visibleTabs: [0, 1], hiddenTabs: [2, 3, 4, 5]};
+      expect(actualIndices).toEqual(expectedIndices);
+    });
+  });
+
+  describe('<Popover />', () => {
+    it('renders a Popover with preferredPosition below when there are hiddenTabs', () => {
+      const mockTabs = [
+        {content: 'Tab 1', id: 'tab-1'},
+        {content: 'Tab 2', id: 'tab-2'},
+        {content: 'Tab 3', id: 'tab-3'},
+        {content: 'Tab 4', id: 'tab-4'},
+        {content: 'Tab 5', id: 'tab-5'},
+        {content: 'Tab 6', id: 'tab-6'},
+      ];
+
+      const tabs = mountWithAppProvider(
+        <Tabs selected={0} tabs={mockTabs} onSelect={noop} />,
+      );
+      tabs.setState({
+        tabWidths: [82, 160, 150, 100, 80, 120],
+        containerWidth: 300,
+      });
+
+      const popover = tabs.find(Popover);
+      expect(popover).toHaveLength(1);
+      expect(popover.prop('preferredPosition')).toBe('below');
+    });
+
+    it('renders a Popover with a button as the activator when there are hiddenTabs', () => {
```

Comment:
This test title seems a but long? I feel like the `renders a Popover` doesn't need to be there as its already being tested above? Also this is within a `<Popover />` describe so arguably you don't need to repeat that anywhere.